### ISSUE TITLE
Authorization OAuth 2.0: Request Authorization Code not working

### ISSUE THREAD

**Issue by jmatelet** [a85/POSTMan-Chrome-Extension#3977]
Authorization OAuth 2.0: enable to request a authorization code token

## App Details:
Postman for Windows
Version 5.5.0
win32 10.0.10586 / x64

## Issue Report:
 1. Did you encounter this recently, or has this bug always been there:
I did not encounter this issue when using the chrome app. This problem is since the windows application.

 2. Expected behaviour:
I should be able to request an Authorization Code token by pressing the Request Token button on the "Get New Access Token" window.
Instead, an empty window is opened.

 3. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app):
No console log... 
 4. Screenshots (if applicable)

the call back url is correctly set to https://www.getpostman.com/oauth2/callback
all other fields are correctly set.
We are able to request a client credential token but not an authorization code. A click on Request Token opens an empty window.

It was working like a charm on the postman chrome app.

**Comment by harryi3t** [comment]
@jmatelet Does your network requires a proxy to work? The request for the modal window does not go through any proxy, so that might be causing issues.

Also, could you share a screenshot of the filled modal (Get new access token)?

**Comment by jmatelet** [comment]
@harryi3t there is no proxy. Requests to the token services (client credential) is working good, but the authorization code doesnt work.
![postman](https://user-images.githubusercontent.com/32544062/34099874-089bbb60-e3e1-11e7-9809-255efeb49669.png)

**Comment by harryi3t** [comment]
@jmatelet Thanks for the details.
Can you make sure once that the callback URL registered on your auth server is "https://www.getpostman.com/oauth2/callback" only

Opening the modal window is almost same as opening a browser window (with few query parameters added). So ideally this should not have issues.

Can you try opening the same in a browser? You need to substitute `AUTH_URL ` and `CLIENT_ID`.
```
{{AUTH_URL}}?response_type=code&state=12344&client_id={{CLIENT_ID}}&scope=resource.READ&redirect_uri=https://www.getpostman.com/oauth2/callback
```
Please post a screenshot of the browser window and the URL also (I am hoping it will contain some error message in a non-standard format).

**Comment by georules** [comment]
I have also observed an issue with postman 5.5.0 (osx) that sounds similar to this one.  I am not sure if related, but our /auth route has multiple stages, and it seems postman closes the login window before it allows the transition to the next page.  This doesn't happen in the chrome version or older versions of postman (5.4.0?)

I note that grant_type doesn't have separators between fields and values, which might explain the invalid_grant response.  (note grant_type=authorization_codecode= ...)

```
POST <<REMOVED>>
content-type: application/x-www-form-urlencoded
authorization: Basic <<REMOVED>>
user-agent: PostmanRuntime/7.1.1
accept: */*
host: <<REMOVED>>
accept-encoding: gzip, deflate
content-length: 196

grant_type=authorization_codecode=<<CODE>>redirect_uri=https://www.getpostman.com/oauth2/callbackclient_id=<<ID>>

HTTP/1.1 400
status: 400
date: Mon, 18 Dec 2017 15:42:57 GMT
server: Apache
x-powered-by: Undertow/1
content-type: application/json
vary: Accept-Encoding
content-encoding: gzip
connection: close
transfer-encoding: chunked
{"error_description":"Code is expired","error":"invalid_grant"}
```

**Comment by jmatelet** [comment]
@harryi3t 
The problem is not coming from my AS authorize service as the following request is successfull 
`{{AUTH_URL}}?response_type=code&state=12344&client_id={{CLIENT_ID}}&scope=resource.READ&redirect_uri=https://www.getpostman.com/oauth2/callback`
From a browser, I can login and I receive the code in the callback response https://app.getpostman.com/oauth2/callback?code=of76sfoPZaJNQ9gCLHdARsOKsZvSiw&state=12344


On the authorization server side, I can see the incoming authorize HTTP  request when I open the request from my browser.
I don't see any incoming HTTP request when I request a token from Postman. It is like the request is never sent by the app.

**Comment by vpzed** [comment]
I'm also having trouble with invalid_grant and grant_type= line not having the & separators.

This looks very wrong:

    grant_type=authorization_codecode=Msomestuff-ce16b4678cderedirect_uri=https://www.getpostman.com/oauth2/callbackclient_id=123456

Should it be more like:

    grant_type=authorization_code&code=Msomestuff-ce16b4678cde&redirect_uri=https://www.getpostman.com/oauth2/callback&client_id=123456

![image](https://user-images.githubusercontent.com/30138020/34551129-4f36ae1a-f0dc-11e7-9c3d-f773de20aded.png)

**Comment by vpzed** [comment]
Over the last couple days I have tried every option I can think of to see if I could get Oauth 2.0 working with Postman 5.5.0 on Windows but no luck.  It was working without issues in a previous version (unfortunately I don't remember the version number) but the Oauth 2.0 screen was different - it was the previous one with the Request access token locally checkbox at the bottom.

I'm open to suggestions on things to try.

**Comment by harryi3t** [comment]
@vpzed 

> This looks very wrong:
grant_type=authorization_codecode=Msomestuff-ce16b4678cderedirect_uri=https://www.getpostman.com/oauth2/callbackclient_id=123456

This should not happen. What are you using to intercept the request?
I tried with a proxy in between ([mitm proxy](http://docs.mitmproxy.org/en/stable/install.html)) and this is what I see

![mitmproxy_-p_8079_--setheader___q_x-proxy_mitm___python3_6_](https://user-images.githubusercontent.com/5207331/34822019-83ba4dee-f6eb-11e7-833b-2cd7feafd8d6.jpg)

The only known bug in Postman for Oauth2 is that if there are multiple Oauth flows going on then we wrongly take the `code` from the first possible redirection.

Could you try starting Postman from console and see the logs (in shell) and post it here.
It logs all the redirections. We can check if in this case we are indeed picking up the wrong code.

**Comment by vpzed** [comment]
@harryi3t I copied that from the "Postman Console" accessed via View > Show Postman Console.

Here is what is shown in Powershell when postman is started via ./postman.exe


	Opening auth login window https://www.bungie.net/en/oauth/authorize?response_
	type=code&state=supersecret&client_id=20895&scope=&redirect_uri=https%3A%2F%2Fwww.getpostman.com%2Foauth2%2Fcallback
	Received redirect on auth login window https://www.bungie.net/en/User/JoinUp?bru=%252Fen%252Foauth%252Fauthorize%253Fresponse_type
	%253Dcode%2526state%253Dsupersecret%2526client_id%253D2085%2526scope%253D%2526redirect_uri%253Dhttps%25253A%25252F%25252F
	www.getpostman.com%25252Foauth2%25252Fcallback
	Received redirect on auth login window https://www.bungie.net/en/User/JoinUp?bru=%252Fen%252Foauth%252Fauthorize%253Fresponse_type
	%253Dcode%2526state%253Dsupersecret%2526client_id%253D2085%2526scope%253D%2526redirect_uri%253Dhttps%25253A%25252F%25252F
	www.getpostman.com%25252Foauth2%25252Fcallback
	Received redirect on auth login window https://www.bungie.net/en/User/SignIn/Xuid?bru=%25252Fen%25252Foauth%25252Fauthorize%25253F
	response_type%25253Dcode%252526state%25253Dsupersecret%252526client_id%25253D2085%252526scope%25253D%252526redirect_uri%2
	5253Dhttps%2525253A%2525252F%2525252Fwww.getpostman.com%2525252Foauth2%2525252Fcallback&flowStart=1
	Received redirect on auth login window https://login.live.com/oauth20_authorize.srf?client_id=00000000013231D&scope=Xboxlive.sign
	in%20Xboxlive.offline_access&response_type=code&redirect_uri=https%3A%2F%2Fwww.bungie.net%2Fen%2FUser%2FSignIn%2FXuid&display=touc
	h&locale=en&state=723118240083371931
	Received redirect on auth login window https://login.live.com/oauth20_authorize.srf?client_id=00000000013231D&scope=Xboxlive.sign
	in%20Xboxlive.offline_access&response_type=code&redirect_uri=https%3A%2F%2Fwww.bungie.net%2Fen%2FUser%2FSignIn%2FXuid&display=touc
	h&locale=en&state=723118240083371931
	Received redirect on auth login window https://login.live.com/oauth20_authorize.srf?client_id=00000000013231D&scope=Xboxlive.sign
	in%20Xboxlive.offline_access&response_type=code&redirect_uri=https%3A%2F%2Fwww.bungie.net%2Fen%2FUser%2FSignIn%2FXuid&display=touc
	h&locale=en&state=723118240083371931
	Received redirect on auth login window https://login.live.com/oauth20_authorize.srf?client_id=00000000013231D&scope=Xboxlive.sign
	in%20Xboxlive.offline_access&response_type=code&redirect_uri=https%3A%2F%2Fwww.bungie.net%2Fen%2FUser%2FSignIn%2FXuid&display=touc
	h&locale=en&state=723118240083371931
	Received redirect on auth login window https://www.bungie.net/en/User/SignIn/Xuid?code=M338322de-0d58-9663-6f6b05e698a7&state
	=723118240083371931
	Matched code/access_token. Closing auth login window now.
	Requesting OAuth 2.0 access token


It stops there.  Then in the View > Show Postman Console window it shows:

	POST
	https://www.bungie.net/platform/app/oauth/token/
	POST /platform/app/oauth/token/
	content-type: application/x-www-form-urlencoded
	user-agent: PostmanRuntime/7.1.1
	accept: */*
	host: www.bungie.net
	accept-encoding: gzip, deflate
	content-length: 212
	grant_type=authorization_codecode=M338322de-0d58-9663-6f6b05e698a7redirect_uri=https://www.getpostman.com/oauth2/callbackclient_id=2085client_secret=c<snip>wMqdoo
	HTTP/1.1 400
	status: 400
	date: Thu, 11 Jan 2018 14:36:46 GMT
	content-type: application/json; charset=utf-8
	content-length: 72
	connection: keep-alive
	set-cookie: __cfduid=d75<snip>
	vary: Origin
	x-bungienext-mid2: 17
	x-bungienext-renderer: Frog Blast the Ventcore
	x-ua-compatible: IE=edge
	p3p: CP="ALL DSP COR PSD PSA TAI ADM CUR SAM OUR IND UNI COM INT CNT PRE"
	access-control-allow-origin: https://www.bungie.net
	x-selfurl: https://www.bungie.net/platform/app/oauth/token/
	x-content-type-options: nosniff
	x-frame-options: SAMEORIGIN
	server: cloudflare
	cf-ray: 3db89c38395f201e-DFW
	{"error":"invalid_grant","error_description":"AuthorizationCodeInvalid"}

Also I do believe there is another redirect in there somewhere because usually after the Xbox Live redirection to enter login credentials, there is another page from Bungie.net that asks the user if they want to authorize the application and shows the permissions being requested by the application.  I do not see that coming up.  I'm not sure where that occurs in the flow exactly so I don't know if it isn't coming up because the process isn't proceeding far enough or if it has something to do with the "redirect bug" you mentioned.

**Comment by jmatelet** [comment]
On my side, I observe a different behavior than the one described by @vpzed 
when I press the Request Token button (flow auth code) I don't see any log in the Postman Console. A blank Postman window is opened and I can see the following line in PowerShell, then nothing happen. the request is never sent to the authorization server :

`Opening auth login window https://192.168.70.130:8089/api/oauth/authorize?response_type=code&state=12344&client_id=40854
f63-0d3e-49ee-ac06-c081fc2c22eb&scope=resource.READ%20write&redirect_uri=https%3A%2F%2Fwww.getpostman.com%2Foauth2%2Fcal
lback`

The exact same request is working fine from a browser.

If I change the Grant Type to Client Credentials, then it is working fine, and I can view the log in Postman Console.

**Comment by harryi3t** [comment]
@vpzed This is exactly the same issue I was talking about.
We captured the `code` from an intermediate URL before it could redirect to the registered redirect URL
```
https://www.bungie.net/en/User/SignIn/Xuid?code=M338322de-0d58-9663-6f6b05e698a7&state
=723118240083371931
```

We have already fixed it internally. This will be shipped with the next Postman release :)

@jmatelet Still we are not sure what might be causing this. You have already mentioned that there are no proxies involved. Could you try any other Oauth2 provider? I am suspecting that this might be due to the server being run locally and that too on https. So there can be certificate related issues.

Please note that any certificates added to postman does **not** get applied to the browser window that opens for Oauth2.

**Comment by jmatelet** [comment]
@harryi3t 
It is working for the client credentials flow which is also using HTTPS with the same server, so I don't think it can be related to https. Plus I didn't had to add any certificates.
Anyway, If I manually create an oauth/authorize request I can reach the server. It doesn't even reach the server with the OAuth2 request token with authorization code grant.

**Comment by ScrappyXII** [comment]
Experiencing the same issue as @jmatelet where the request does not reach the server with an OAuth2 request with authorization grant. Empty pop up occurs and nothing happens.

**Comment by taftse** [comment]
I'm having the exact same issue from looking at the logs on both the server and within postman it looks like the postman window is not saving the session state 

when I login its supposed to show me the screen to authorize postman but it redirects me back to the login screen because there is no valid session 

however I'm not having this issue with the old chrome extension or if I visit the page in a normal browser window

**Comment by harryi3t** [comment]
@taftse @ScrappyXII @jmatelet 
Is your auth provider public? Can I register myself and this it out?
If not, will it be possible for you to share me some temporary credential for testing?

**Comment by taftse** [comment]
@harryi3t I have sent you an email ( to the gmail address listed in your profile) with the details

**Comment by harryi3t** [comment]
@taftse I have received your email. I will register and try it out.
One thing I noticed was that you are using a different callback URL in the new native app as compared to the one you were using in chrome app. Was it intentional? Did you change the registered callback URL in the auth server?

```
https://www.getpostman.com/oauth2/callback // using in chrome app
https://app.getpostman.com/oauth2/callback // using in native app
```

Can you try once using the old callback URL in the native app?

**Comment by ScrappyXII** [comment]
@harryi3t I am using an internal Oracle test system so am unable to provide access to that. However, you may be able to try registering a trial Oracle Cloud account: https://cloud.oracle.com/tryit?intcmp=ocom-hp
And accessing the Identity service which provides OAuth2 support.

**Comment by taftse** [comment]
@harryi3t you can ignore my issue it was indeed the callback url I cant believe missed that sorry for wasting your time

**Comment by harryi3t** [comment]
@taftse Not able register. Throwing UnexpectedValueException
![image](https://user-images.githubusercontent.com/5207331/36668279-c926559a-1b16-11e8-87fe-d2e0f68a2afa.png)

I have mailed you the full screenshot of the error.
<hr>
@ScrappyXII  I also tried registering an account with Oracle Cloud but could not succeed.
The phone verification fails always :( Server is returning 400.

![oracle_cloud_account_signup](https://user-images.githubusercontent.com/5207331/36668713-2c772fc4-1b18-11e8-9797-9b588b97e8d0.png)

![image](https://user-images.githubusercontent.com/5207331/36668842-94ef1d64-1b18-11e8-9fdd-958b1ce12ffd.png)

**Comment by ScrappyXII** [comment]
@harryi3t  please try again i did not see this issue. If problem persists for you, would suggest contact cloud support for resolution as i dont have visibility into that.

**Comment by jmatelet** [comment]
sadly... the initial problem described in this ticket still exists in the last version of the postman application. 
it was working great with the checkbox "request token locally" of the chrome extension...
please consider this issue? can you confirm that it is working for you?

**Comment by nwchandler** [comment]
I am experiencing the same behavior described by @jmatelet on:

Postman for Mac
Version 6.1.2
OS X 15.6.0 / x64

An empty window opens, and there is nothing in the log. There are also no records in my authorization server's logs, as though it never receives any communication from Postman.

**Comment by zpq2004** [comment]
I am experiencing the same behavior described by @jmatelet on:
Postman for Mac
Version 6.0.10
macOS High Sierra version 10.13.3

**Comment by bobvanderlinden** [comment]
I ran into issues with OAuth2 and Postman as well. Not 100% sure this is the same cause as the OP is experiencing. I'm seeing the following POST body towards the `token` url:

```
grant_type=authorization_codecode=N3ZIYkZlQ1BNSzFMd3NkcXZ4bDkzenRvQnZLejFhSHJpdWd4YzF1Q0pOQzkrOG90K1J1a0IrQU1KRU1vc3pIZDRQcFFwQ1RXTDhncElxSzNaVlp0NVBSYmVtMUwyaFdhQ2tMTHlkSk52Qm85ZVVQUFdmWWhCVFJwQWtxbi9VTHFzU0h2bTk1V05hYWNlZE8wZTdyZ2VsYUZ0Z2g2d251N2VnSWJyL0xHUHVuRVl4S0tLOVdFNU5lemMrR1dJSGxHUER5Qnd5NDZxcXhVVGdWYWZsSldwcC9qOTllVWFwSEV3TW9jcGc1bmx0RT0tLTVjUnYyUmZ5NkNHaldQdkpnZzg3eXc9PQ==--927108cbb5b7a1babdb051051b64ca4ea4a79a47redirect_uri=http://bogus/client_id=my-client-id
```
The problem seems 3-fold:

1. The RAW representation of request bodies is leaving out `&`, which makes this issue extra confusing. The actual call itself is sending `&` correctly.
2. The problem (in my case at least) is that the `state` is not being send to the token URL. It is being sent to the authorize call, which makes the code required for the token URL. That's why the all is failing in my case.
3. When the `state` is not set in the UI, Postman will still send `&state=` to the authorize call, which means the state is set to empty (but still specified!). Because it is specified (as empty) it is required to be empty in the token call.

For reference, in is the API documentation of Fitbit it says about the state parameter for the authorize call:

> Provides any state that might be useful to your application when the user is redirected back to your application. This parameter will be added to the redirect URI exactly as your application specifies in the authorization request. Fitbit strongly recommend including an anti-forgery token in this parameter and confirming its value in the redirect to mitigate against cross-site request forgery (CSRF).

Source: https://dev.fitbit.com/build/reference/web-api/oauth2/#authorization-page

Then it says about the state parameter in the token call:

> Required if specified in the redirect to the authorization page. Must be an exact match.

Source: https://dev.fitbit.com/build/reference/web-api/oauth2/#access-token-request

Also, Insomnia does not have this behaviour and handles the call correctly.

**Comment by harryi3t** [comment]
@jmatelet Are you still facing this issue with the latest version of Postman (6.1.4)?
Since the original issue was about seeing a blank screen, if that is not happening now we can close out this issue.
@bobvanderlinden and others, if your issue is different than the original issue please open a separate issue.

**Comment by jmatelet** [comment]
Hi @harryi3t 
As I said in my previous comment, I am still facing the issue.
I am still unable to request an Authorization Code  (display a blank screen as described) with postman app 6.1.4

The postman app needs a "request locally" checkbox that was removed since postman chrome extension... it is working very good with postman chrome extension but not with the postman app.

**Comment by harryi3t** [comment]
@jmatelet Thanks for the response. Is the auth provider public or is there any way I can test on this auth provider.

> The postman app needs a "request locally" checkbox that was removed since postman chrome extension

The checkbox was removed because now it's the default behavior.

Update:
Earlier when you tried opening the auth URL in a browser and when you successfully got the code, did the browser at any point showed a prompt for login asking for username/password?

**Comment by jmatelet** [comment]
@harryi3t 
yes, from the browser it showed a prompt for login username/password.

I am trying to debug and it seems I am getting blocked by the HTTPS handshake using postman app... the app seems to shut down the connection. maybe a cipher issue on postman side...

**Comment by harryi3t** [comment]
@jmatelet I think I might know the issue here. Postman uses an electron window to open the auth URL. The opened window is similar to a browser window but cancels all authentications prompt.

Can you tell me which auth is the endpoint protected by? (Usually, when there is a prompt in the browser, it's one of basic, digest or NTLM auth)
Electron has added support for only basic auth https://github.com/electron/electron/pull/3250

Related Issue: https://github.com/postmanlabs/postman-app-support/issues/4246

**Comment by jmatelet** [comment]
It is not the same issue as the login page is not even displayed. This login page is not using any particular http auth... it is a simple form.

It is probably an HTTPS handshake issue. The authorization server is not even processing the HTTPS request as the HTTPS handshake is closed before anything is processed on server side.
(negotiated SSL cipher "ECDHE-RSA-AES128-GCM-SHA256")

**Comment by M3yo** [comment]
I have the exact same symptoms as @jmatelet, I'm using Postman v6.1.4, I want to request an OAuth2 token and : 

- Blank page when I request token
- No trace on console, or auth server, only "Opening auth login window https://...." when I open from a shell. (url working in my browser)
- Working with postman chome app

I'm using Keycloak as OAuth2 provider, and we are (for tests only) using a self-signed certificate. 
So I decided to try on a different Keycloak server running in http only. And now it works, I can log through and use my token.

So this seems to be related with SSL  -> I've already turned off "SSL certificate verification" in settings but this doesn't seem to have any effect (for token requesting that is.)

**Comment by harryi3t** [comment]
@jmatelet Are you also using a self-signed certificate? Can you confirm that using only HTTP works?

**Comment by ScrappyXII** [comment]
@harryi3t the service i am using has a self signed cert and I am still experiencing this same/similar issue as @jmatelet and @M3yo. I can’t validate that only HTTP works as the service does not expose that.

**Comment by jmatelet** [comment]
I am using a certificate that is signed by the CA of my company. 
I tried using HTTP and now postman successfully send the /authorize request to the server.
So this problem is related with SSL

**Comment by jmatelet** [comment]
@harryi3t can you confirm that this is a SSL problem ? 

I can't make it work using HTTPS using the last version of postman... I used both autosign certificates and CA signed certificates. 
Is there a way to configure postman server certificate validation?

**Comment by harryi3t** [comment]
@jouzeroff I have confirmed that this is indeed an SSL verification issue.

@M3yo 
> So this seems to be related with SSL -> I've already turned off "SSL certificate verification" in settings but this doesn't seem to have any effect (for token requesting that is.)

The setting `SSL certificate validation` is only being applied to request sending.
For now I will be fixing this by making sure that the setting applies to OAuth-2 window as well.

**Comment by thomaslazar** [comment]
Any ETA on this issue? I wasted about half a days worth trying to find out why postman wouldn't show anything in the window when clicking "Request Token".
First I thought my ADFS server was bonkers but it turned out it was exactly because it doesn't honor the ignore SSL Cert setting. 
Was using Postman 6.6.1

**Comment by jvalteren** [comment]
Same here. I'm trying to login to Exact Online using oauth2, which apparently first does a POST which redirects to GET which then should respond with a location header that has the access token 'code' parameter in the query string (it does in a normal browser).
However, in the DevTools (Current Shell) view in Postman 6.7.0, it show the request in red with status canceled.

**UPDATE (Workaround):** 
Using the same DevTools, clearing the site data resolved the issue! Can be found under 'Application > Clear storage > Clear site data'. (Thanks for the tip @aldegoeij!)

Probably removing the cookies in the 'Manage cookies' dialog (click on Cookies under the Send button) will have the same effect.

**Comment by harryi3t** [comment]
@jouzeroff You mentioned that when you try it in the browser it works, also it should a prompt for username/password. Could you share a screen recording of that (please use incognito mode)?
I just wanted to confirm whether the prompt is a native browser prompt or actually a UI for login flow.

We are tracking an issue where if the prompt is a native browser prompt, then it is skipped in Postman due to a limitation of the underlying platform we use (Electron). If I can confirm that this is the issue then we can work in that direction.

**Comment by harryi3t** [comment]
@jouzeroff @M3yo @thomaslazar 
As mentioned in my previous comment, we have added a workaround so that now disabling the `SSL certificate validation` from the settings modal will turn it off for the Oauth2 window as well. This was released with version 6.7.4. Could you verify this and let us know.

Note: The self-signed certificates do not work since the platform on which Postman is built (Electron) does not support reading/resolving the certificates from the key-chain (or the equivalent credentials store on other operating systems).

**Comment by cwarden** [comment]
I'm not sure if it's the same issue(s) reported here, but when using the Authorization Code Grant Type, Postman sends the `client_id` when exchanging the code for a token, which violates RFC 6749.  This will break the code exchange against servers that validate the form variables.  (I ran into this with Salesforce.)

There was a similar bug in the Go oauth2 library.  The fix may be helpful for reference: https://github.com/golang/oauth2/commit/4464e7848382e6f39db55097f70e1460bdf944b3

**Comment by farsh77** [comment]
I am now seeing this exact issue with version 7.3.3. Did not have this issue with previous version.

**Comment by ClaysonIO** [comment]
I'm also seeing this issue with Version 7.3.2.

Hitting 'Request Token' does open a popup and walk through the authorization flow, but displays a blank page when it arrives at `https://www.getpostman.com/oauth2/callback`, without any flow to get the code back into Postman.

**Comment by harryi3t** [comment]
@ClaysonIO This issue should be fixed in the latest version of Postman 7.3.4. Can you try this after updating your app?

**Comment by ClaysonIO** [comment]
@harryi3t That did it, thanks!

**Comment by farsh77** [comment]
Just updated to 3.4 and it is fixed now. Thanks.

**Comment by harryi3t** [comment]
@ClaysonIO @farsh77 Thanks for the confirmation.
Closing this ticket.

**Comment by BeingSujeetK** [comment]
@harryi3t I'm also facing this issue in Postman for Windows Version 7.3.6. It was working perfectly in earlier versions until it got auto updated few weeks back.

When I click on Request Token button., the flow stops before requesting user credentials with a blank screen like below.
If I restart Postman, the flow continues till the user credentials is submitted and then gets stuck on a blank screen like below.

![image](https://user-images.githubusercontent.com/16061402/62856903-c34ad380-bd13-11e9-9554-9b055bae7e69.png)

Thanks in advance!

**Comment by ankurdengla** [comment]
Hey @BeingSujeetK ! Could you let us know the following details - 

- `callback url` that you are using to obtain the access token

- Last version where the flow was working


 We are already tracking a case where OAuth2 flow doesn't work if there is a custom protocol used in the `callback url` here - https://github.com/postmanlabs/postman-app-support/issues/7032

**Comment by BeingSujeetK** [comment]
@ankurdengla1996 Will follow up in the new thread. That seems more relevant.

**Comment by mcandre** [comment]
Hi,

I am also getting `invalid_grant` error. Just a few months ago, Get New Access Token was working perfectly with Okta.

**Comment by sharebright** [comment]
Hi, 
I am using version 7.13.0 of postman and using IdentityServer4 and having the same issue with 0Auth2 flow as reported here - https://community.getpostman.com/t/using-postman-to-test-identityserver4-oauth-authentication-code-flow/9524

**Comment by shamasis** [comment]
https://github.com/postmanlabs/postman-app-support/issues/7700#issuecomment-626273229 this comment should allow to use browser flow to ensure any such issue is addressed.
